compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -125,7 +125,8 @@ object DesugarEnums {
   /** A creation method for a value of enum type `E`, which is defined as follows:
    *
    *   private def $new(_$ordinal: Int, $name: String) = new E with scala.runtime.EnumValue {
-   *     def $ordinal = $tag
+   *     private[this] def $ordinal = _$ordinal // if deriving from jl.Enum
```

Comment:
Why is this private method needed at all?

Reply:
At least in the master branch the ordinal method was used to store the ordinal as an int literal, and without private $ordinal that literal goes away, we could use a symbol attachment on the class instead maybe

Reply:
now desugaring does not override toString or create $ordinal for java enum implementations